service/src/lib.rs

```diff
@@ -282,7 +265,7 @@ extern crate shuttle_codegen;
 /// struct MyState(PgPool);
 ///
 /// #[shuttle_service::main]
-/// async fn rocket(#[shared::Postgres] pool: PgPool) -> ShuttleRocket {
+/// async fn rocket(#[shuttle_shared_db::Postgres] pool: PgPool) -> ShuttleRocket {
```

Comment:
We should document how these attributes work if we haven't already. AFAICT, the proc macro just expects these to be type addresses to something implementing `ResourceBuilder<T>` where `T` is the type of the argument you want to instantiate.

If we document this, people could write their own hook to grab the `Factory` and manage resources however they want.

Reply:
Document is now up at https://github.com/shuttle-hq/shuttle/pull/273/files#diff-a981df14dcb291acc010b3484e3911046bb9f568ebd28ed7c6c2bc4de23aa1faR304